src/TNode.h

```diff
@@ -0,0 +1,54 @@
+//
+// Created by shvmo on 29/04/2020.
+//
+
+#ifndef DATASTRUCTURESWET1_TNODE_H
+#define DATASTRUCTURESWET1_TNODE_H
+
+
+#include <memory>
+
+using std::shared_ptr;
+
+template<class Key, class Info>
+class TNode {
+public:
+    TNode();
+
+    ~TNode();
+
+    Key &getKey();
+
+    Info &getInfo();
+
+    shared_ptr<Info> getInfoPointer() const;
+
+    shared_ptr<TNode<Key, Info>> getLeft() const;
+
+    shared_ptr<TNode<Key, Info>> getRight() const;
+
+    void setKey(const Key &newKey);
+
+    int getHeight() const;
```

Comment:
seems like you wrote getters and then the setters so you might as well move it to the getters part

Reply:
What?

Reply:
fiixed